lib/Ast_to_Mir.ml

```diff
@@ -251,6 +251,7 @@ let pull_tvdecls = function
    * model -> no constraints allowed (not even corr_matrix et al); not visible
    * gq -> declared and checked in write_array, shows up as param in some methods
 
+   Local vs. Global vardecls
    So there are "local" (i.e. not top-level; not read in or written out anywhere) variable
    declarations that do not allow transformations. These are the only kind allowed in
    the model block. There are also then top-level ones, which are the only thing you can
```

Comment:
Basically, I think the rule is that variable declarations are always global unless these occur inside a block statement or in the model-block. You can check on a `Ast.VarDecl` whether it is global by inspecting the `is_global` flag.

Reply:
Yep, that's right. I can add a note about the fact that if they're nested in a Block they are also local.